### ISSUE TITLE
Anomalous unresolved symbol errors in `reg-event-fx` macro body

### ISSUE THREAD

**Issue by kelvinqian00** [clj-kondo/clj-kondo#1704]
**Version and Editor**

Initially discovered via **Calva 2.0.276** on VSCode. Reproduced on **clj-kondo v2022.04.25** using the CLI tool.

**Macro usage**

The macro involved is the `reg-event-fx` macro from the `re-frame` library, which is one of the libraries whose macros is supported by clj-kondo by default.

**Problem**

A variable defined in an anonymous function passed to `reg-event-fx` will be considered as an unresolved symbol error by clj-kondo, if it is contained within a vector used by `:dispatch` in an `:fx` value.

**Reproduction**

The following is a minimal example in the form of a toy event subscription:
```clojure
(ns example
  (:require [re-frame.core :as reframe]))

(re-frame/reg-event-fx
 :foo/bar
 (fn [_ [my-arg]]
   {:fx [[:dispatch [my-arg]]]})) ; The unresolved symbol error occurs on `my-arg`
```

The error appears regardless of how many vectors either `my-arg` or `:dispatch [my-arg]` is nested in, so long that `my-arg` appears in at least one vector.

As such, the following _does not_ present an error, since `my-arg-vec` is itself not wrapped in any vectors:

```clojure
(re-frame/reg-event-fx
 :foo/bar
 (fn [_ my-arg-vec]
   {:fx [[:dispatch my-arg-vec]]})) ; No error
```

The error also only occurs with the `:dispatch` keyword:
```clojure
(re-frame/reg-event-fx
 :foo/bar
 (fn [_ [my-arg]]
   {:fx [[:foo [my-arg]]]})) ; No error (but incorrect re-frame code)
```

When running `clj-kondo --lint example.cljs` (assuming no other errors), one gets the following:
```
example.cljs:7:22: error: Unresolved symbol: my-arg
linting took [xx]ms, errors: 1, warnings: 0
```

**Config**

The following is our current `config.edn` file, but the error occurs if no such file is present as well:
```clojure
{:lint-as
 {secretary.core/defroute clojure.core/defn}}
```

**Expected behavior**

All variables that are defined in anonymous functions passed to `reg-event-fx` should not result in clj-kondo errors.

**Comment by borkdude** [comment]
cc @benedekfazekas

**Comment by borkdude** [comment]
@benedekfazekas I think a check is missing here whether the event-id is a keyword or not:

https://github.com/clj-kondo/clj-kondo/blob/7a711ef748c401f3115e5a7823a7fa1b25125617/src/clj_kondo/impl/analyzer/re_frame.clj#L74

When it's not a keyword, analysis should probably be omitted there?

**Comment by borkdude** [comment]
@kelvinqian00 A temporary workaround:

``` clojure
(re-frame/reg-event-fx
 :foo/bar
 (fn [_ [my-arg]]
   {:fx [[:dispatch (vector my-arg)]]}))
```

or:

``` clojure
(ns example
  (:require [re-frame.core :as re-frame]))

(re-frame/reg-event-fx
 :foo/bar
 (fn [_ [my-arg]]
   {:fx [[:dispatch #_:clj-kondo/ignore [my-arg]]]}))
```

**Comment by kelvinqian00** [comment]
Appreciate the workaround suggestions. I'm currently using the following:
```clojure
#_{:clj-kondo/ignore [:unresolved-symbol]}
(re-frame/reg-event-fx
 :foo/bar
 (fn [_ [my-arg]]
   {:fx [[:dispatch [my-arg]]]})) ; Error is suppressed
```
but it seems like your second suggestion is more precise.

**Comment by borkdude** [comment]
Fixed with #1707
